tracer/build/_build/Build.Steps.cs

```diff
@@ -486,15 +512,16 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
         .After(CompileNativeSrc, PublishManagedTracer)
         .Executes(() =>
         {
-            foreach (var (arch, folder, extension) in GetMacOsArchitectureAndExtension())
-            {
-                // Copy Native file
-                CopyFile(
-                    NativeTracerProject.Directory / "build" / "bin" / $"{NativeTracerProject.Name}.{arch}.{extension}",
-                    MonitoringHomeDirectory / folder / $"{NativeTracerProject.Name}.{extension}",
-                    FileExistsPolicy.Overwrite,
-                    true);
-            }
+            // Create universal shared library with all archs in a single file
+            var lstNativeBinaries = OsxArchs
+                                   .Select(arch => NativeTracerProject.Directory / "build" / "bin" / $"{NativeTracerProject.Name}.{arch}.dylib");
+
+            var destination = $"./{NativeTracerProject.Name}.dylib";
+            var destFolder = MonitoringHomeDirectory / "osx";
+            Console.WriteLine($"Creating universal binary for {destination}");
+            EnsureExistingDirectory(destFolder);
+            var strNativeBinaries = string.Join(' ', lstNativeBinaries);
+            Lipo.Value(arguments: $"{strNativeBinaries} -create -output {destFolder}/{destination}");
```

Comment:
nit: Could we instead create the universal binary in the compile step? And then the publish step would just copy it to the `MonitoringHomeDirectory / "osx"` folder?

Reply:
done in [62fda46](https://github.com/DataDog/dd-trace-dotnet/pull/3454/commits/62fda46102ad930801ed65d514e8ed655da5dea3)